packages/salesforcedx-vscode-core/test/vscode-integration/commands/forcePackageInstall.test.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import * as vscode from 'vscode';
+import {
+  ForcePackageInstallExecutor,
+  SelectInstallationKey,
+  SelectPackageID
+} from '../../../src/commands/forcePackageInstall';
+import { nls } from '../../../src/messages';
+
+// tslint:disable:no-unused-expression
+describe('Force Package Install', () => {
+  describe('SelectPackageID Gatherer', () => {
+    const EVENT_CANCEL = 'CANCEL';
+    const EVENT_CONTINUE = 'CONTINUE';
+    const TEST_PACKAGE_ID = 'testPackageID';
+    let inputBoxSpy: sinon.SinonStub;
+
+    beforeEach(() => {
+      inputBoxSpy = sinon.stub(vscode.window, 'showInputBox');
+    });
+
+    afterEach(() => {
+      inputBoxSpy.restore();
+    });
+
+    it('Should return cancel if package ID is undefined', async () => {
+      inputBoxSpy.onCall(0).returns(undefined);
+      const gatherer = new SelectPackageID();
+      const response = await gatherer.gather();
+      expect(inputBoxSpy.callCount).to.equal(1);
+      expect(response.type).to.equal(EVENT_CANCEL);
+    });
+
+    it('Should return cancel if packageId is empty string', async () => {
```

Comment:
Instead of cancelling maybe this should be caught by the validation

Reply:
Yeah, I think I need to update this test now that I've added the command validation

Reply:
Nevermind, this test is still valid.